corehq/apps/cloudcare/static/cloudcare/js/formplayer/js/apps/config/storage/localstorage.js

```diff
@@ -0,0 +1,53 @@
+/*global FormplayerFrontend */
+
+FormplayerFrontend.module("Entities", function (Entities, FormplayerFrontend, Backbone, Marionette, $, _) {
+    var findStorageKey = function (entity) {
+        // use a model's urlRoot value
+        if (entity.urlRoot) {
+            return _.result(entity, "urlRoot");
+        }
+        // use a collection's url value
+        if (entity.url) {
+            return _.result(entity, "url");
+        }
+
+        throw new Error("Unable to determine storage key");
+    };
+
+    var storageCache = {};
+    var getStorage = function (key) {
+        var storage = storageCache[key];
+        if (storage) {
+            return storage;
+        }
+        var newStorage = new Backbone.LocalStorage(key);
+        storageCache[key] = newStorage;
+        return newStorage;
+    };
+
+    var StorageMixin = function (entityPrototype) {
+        var storageKey = findStorageKey(entityPrototype);
+        return {localStorage: getStorage(storageKey)};
+    };
+
+    var getEntity = function (constructorString) {
+        var bits = constructorString.split("."),
```

Comment:
`bits` is pretty ambiguous given the parameter is also `constructorString` if it really is that generalized, a comment would be helpful